docs/JOURNAL_FILE_FORMAT.md

```diff
@@ -518,7 +517,10 @@ the format indicated by the **compact** struct instead of the **regular** struct
 The **realtime** field is stored as a 32-bit offset to the **head_entry_realtime**
 field in the header. Similarly, The **monotonic** field is stored as a 32-bit
 offset to the **head_entry_monotonic** field in the header. The **seqnum** field is
-stored as a 32-bit offset to the **head_entry_seqnum** field in the header.
+stored as a 32-bit offset to the **head_entry_seqnum** field in the header. The
+boot ID is not stored per entry anymore and is always identical to the
+**boot_id** field in the header. As a consequence, journal files written in
+compact mode are limited to a single boot ID.
```

Comment:
Hmm. I have to say that this seems like a potentially bad tradeoff. Some machines reboot regularly, and this will force rotation on every reboot. What about a slightly different approach: in the header, create an array of up to 16 boot_ids, and a corresponding array of seqnums. When looking up the boot_id for a seqnum, find the entry in the second array with the highest entry smaller than the seqnum at hand, and return the boot_id in the same position. This would give essentially the same size savings without forcing rotation…